tests/integration_tests/functional/test_metrics.py

```diff
@@ -33,3 +35,55 @@ def test_flush_metrics(test_microvm_with_api):
         assert microvm.api_session.is_status_no_content(response.status_code)
     lines = metrics_fifo.sequential_reader(how_many_flushes)
     assert len(lines) == how_many_flushes
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="not yet implemented on aarch64"
+)
+def test_stop_and_flush_metrics(test_microvm_with_atkbd):
+    """Check if stop/reboot flush the metrics."""
+    microvm = test_microvm_with_atkbd
+    microvm.spawn()
+    microvm.basic_config()
+
+    # Configure metrics system.
+    metrics_file_path = os.path.join(microvm.path, 'metrics_file')
+    metrics_file = log_tools.File(metrics_file_path)
+
+    response = microvm.metrics.put(
+        metrics_fifo=microvm.create_jailed_resource(metrics_file.path)
+    )
+    assert microvm.api_session.is_status_no_content(response.status_code)
+
+    microvm.start()
+    time.sleep(2)
```

Comment:
I don't think this is robust code, sleeping for two seconds is arbitrarily too long or too short depending on the machine this test runs and/or how loaded the machine is.

It would be better to use a solution that deterministic-ally finds when a microVM has booted, then continues running the test.